exporter/trace/zipkin/zipkin.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2017, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package zipkin contains an exporter for Zipkin.
+//
+// Example:
+//
+// 	import (
+// 		zk "github.com/openzipkin/zipkin-go"
+// 		"go.opencensus.io/trace/adaptor/zipkin"
+// 	)
+//	...
+//		localEndpoint, err := zk.NewEndpoint("my server", myHostPort)
+// 		if err != nil {
+// 			log.Print(err)
+// 		}
+// 		exporter, err := zipkin.NewExporter("http://localhost:9411/api/v2/spans", localEndpoint)
+// 		if err != nil {
+// 			log.Print(err)
+// 		} else {
+// 			trace.RegisterExporter(exporter)
+// 		}
+package zipkin
+
+import (
+	"encoding/binary"
+	"strconv"
+	"strings"
+
+	"github.com/openzipkin/zipkin-go/model"
+	"github.com/openzipkin/zipkin-go/reporter"
+	"github.com/openzipkin/zipkin-go/reporter/http"
+	"go.opencensus.io/trace"
+)
+
+// Exporter is an implementation of trace.Exporter that uploads spans to a
+// Zipkin server.
+type Exporter struct {
+	reporter      reporter.Reporter
+	localEndpoint *model.Endpoint
+}
+
+// NewExporter returns an implementation of trace.Exporter that uploads spans
+// to a Zipkin server.
+//
+// zipkinURL is the address of the Zipkin server.
+//
+// localEndpoint sets the local endpoint of exported spans.  It can be
+// constructed with github.com/openzipkin/zipkin-go.NewEndpoint, e.g.:
+// 	localEndpoint, err := NewEndpoint("my server", listener.Addr().String())
+// localEndpoint can be nil.
+//
+// opts can be used to pass options to the underlying Zipkin http reporter
+// library.
+func NewExporter(zipkinURL string, localEndpoint *model.Endpoint, opts ...http.ReporterOption) (*Exporter, error) {
```

Comment:
zipkinURL can default to the localhost:[defaultport]

Reply:
we usually make the zipkinUrl fully qualified due to people doing it over https or remounting it, or clones

Reply:
The http reporter requires it to be fully qualified.